src/pubtools/_marketplacesvm/cloud_providers/ms_azure.py

```diff
@@ -85,14 +85,16 @@ def destinations(self) -> Set[Any]:
 class AzureProvider(CloudProvider[VHDPushItem, AzureCredentials]):
     """The Azure marketplace provider."""
 
-    def __init__(self, credentials: AzureCredentials) -> None:
+    def __init__(self, credentials: AzureCredentials, **kwargs) -> None:
         """
         Create an instance of AzureProvider.
 
         Args:
             credentials (AzureCredentials)
                 credentials to use the Azure API.
         """
+        default_allow_draft = os.environ.get("AZURE_ALLOW_DRAFT_PUSH", "false").lower() == "true"
```

Comment:
ENV variables are fine for fine tuning some parameters  (integer string) when running from the service (rcm-pub)

Local execution is usually done by human beings. and Command line parameters are more user friendly 
that ENV variables. Ideally with `--azure-` prefix :-)

That way we would be able to support other cloud specific features.
e.g.
* --azure-{break/stop}-before-publish
* --azure-{break/stop}-before-preview
 ...

Reply:
Sure!

I'll patch this accordingly

Reply:
@lslebodn Implemented on commit 6f00877

Reply:
Note that since the cmdline args comes from the `CloudService`, which may instantiate any `CloudProvider` (azure, aws) we need to always set the `kwargs` in the `CloudProvider` constructors hence I've added it on `AWSProvider`